redash/models/__init__.py

```diff
@@ -1210,7 +1210,15 @@ def copy(self, dashboard_id):
             "visualization_id": self.visualization_id,
             "dashboard_id": dashboard_id,
         }
+    
+@generic_repr("id", "widget_id")
+class metrWidget(TimestampMixin, BelongsToOrgMixin, db.Model):
+    id = primary_key("metrWidget")
+    widget_id = Column(key_type("Widget"), db.ForeignKey("widgets.id"))
+    widget = db.relationship(Widget,  backref=backref("metr_widget", cascade="delete"),uselist=False)
```

Comment:
To reference my other comment about the relationship in the serializer. Looks like you should do this:

```suggestion
    widget = db.relationship(Widget,  backref=backref("metr_widget", cascade="delete", ,uselist=False))
```

Reply:
@thiagogds Thanks a lot for your feedback 

Actually, I tried also that and it was not working but I believe that I was using it the wrong way.

...because i did not want to touch the `Widget` class and apparently I have to, because according to [ the documentation](https://docs.sqlalchemy.org/en/13/orm/basic_relationships.html#one-to-one) , the way we link the models to each other should happen as follow:
- add a `relationship` in one , defining this `uselist` thing (should be done `Widget` table here to avoid any migration for it because relationship alone does not provoke any change in model migration)
- add a `relationship` and a `foreign key` in the other one (should be done in `metrWidget`)

It seems like sqlalchemy does not enforce the definition of the `relationship` in the first model (Widget) but not defining it does not seem correct and i am afraid it would affect how we fetch data in the futur so i will add it.

I imagine that by fixing this... all our issues related to how these two models relate will be fixed...change in progress

Reply:
Update:
Sorry, Apparently `bacref` in `db.relationship` can be used from one side but `back_populates` needs to be defined from both side , so i am reconsidering what i said earlier and collecting more information

Reply:
thanks, please find update in [this same commit mentioned previously](https://github.com/metr-systems/redash/pull/4/commits/dde3bb8293de5e68f3bb2b8a313f83c158e51f9a)